hydromt/workflows/to_response_units.py

```diff
@@ -0,0 +1,160 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+from pathlib import Path
+import logging
+from typing import Tuple, Union, Optional
+from .. import flw
+import geopandas as gpd
+import hydromt
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["make_ds_mask", "hydrography_to_basins","create_response_unit_ds",
+           "ru_geometry_to_gpd", "fracs", "ds_class_mode"]
+
+
+def make_ds_mask(ds, gdf, col_name='value'):
```

Comment:
I think the `xr.Datasset.raster.geometry_mask` function is what you are looking for.
https://deltares.github.io/hydromt/latest/_generated/hydromt.DataArray.raster.geometry_mask.html

Reply:
I think this function is no longer used, will check and delete.